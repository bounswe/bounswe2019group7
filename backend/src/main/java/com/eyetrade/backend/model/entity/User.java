package com.eyetrade.backend.model.entity;

import com.eyetrade.backend.constants.Role;
import com.eyetrade.backend.constants.PrivacyType;
import lombok.Data;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import java.util.UUID;

import static com.eyetrade.backend.constants.GeneralConstants.ID_LENGTH;

/**
 * Created by Emir GÃ¶kdemir
 * on 12 Eki 2019
 */

@Data
@Entity
@Table(name="users", uniqueConstraints = {@UniqueConstraint(columnNames = {"email"})})
public class User {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO,generator = "system-uuid")
    @GenericGenerator(name = "system-uuid", strategy = "uuid2")
    @Column(name = "id", length = ID_LENGTH)
    private UUID id;

    @NotNull
    @Column(name = "email")
    @Email
    private String email;

    @Column(name = "pword")
    @NotEmpty(message = "Please provide your password")
    private String pword;

    @Column(name = "name")
    private String name;

    @Column(name = "surname")
    private String surname;

    @Column(name = "phone")
    private String phone;

    @NotNull
    @Column(name = "role")
    @Enumerated(EnumType.STRING)
    private Role role;

    @Column(name = "iban")
    private String iban;

    @Column(name = "identity_no")
    private String identityNo;

    @Column(name = "city")
    private String city;

    @Column(name = "location_x")
    private String locationX;

    @Column(name = "location_y")
    private String locationY;

    @Column(name = "confirmed")
    private boolean confirmed;

    @NotNull
    @Column(name = "privacy_type")
    @Enumerated(EnumType.STRING)
    private PrivacyType privacyType;
}