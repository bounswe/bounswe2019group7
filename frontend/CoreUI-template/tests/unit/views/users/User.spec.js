import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import BootstrapVue from 'bootstrap-vue'
import User from '@/views/users/User'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

localVue.use(BootstrapVue)

describe('User.vue', () => {
  it('has a name', () => {
    expect(User.name).toMatch('User')
  })
  it('has a created hook', () => {
    expect(typeof User.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof User.data).toMatch('function')
    const defaultData = User.data()
    expect(defaultData.fields).toEqual([{key: 'key'}, {key: 'value'}])
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(User, {
      localVue,
      router
    })
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is User', () => {
    const wrapper = shallowMount(User, {
      localVue,
      router
    })
    expect(wrapper.is(User)).toBe(true)
  })
  it('renders props.caption when passed', () => {
    const caption = 'User id:'
    const wrapper = mount(User, {
      propsData: { caption },
      localVue,
      router
    })
    expect(wrapper.find('div.card-header').text()).toMatch(caption)
  })
  it('should have methods', () => {
    const wrapper = shallowMount(User,{
      localVue,
      router
    })

    expect(typeof User.methods.goBack ).toEqual('function')
    expect(wrapper.vm.goBack()).toBeUndefined()
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(User, {
      localVue,
        router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
