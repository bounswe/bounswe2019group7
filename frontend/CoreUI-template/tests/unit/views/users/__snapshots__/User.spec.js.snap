// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User.vue renders correctly 1`] = `
<b-row-stub
  tag="div"
>
  <b-col-stub
    cols="12"
    lg="6"
    tag="div"
  >
    <b-card-stub
      bodytag="div"
      footertag="div"
      headertag="div"
      no-header=""
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <template>
        
        User id:  
      
      </template>
       
      <b-table-stub
        apiurl=""
        currentpage="1"
        emptyfilteredtext="There are no records matching your request"
        emptytext="There are no records to show"
        fields="[object Object],[object Object]"
        fixed="true"
        items="[object Object]"
        labelsortasc="Click to sort Ascending"
        labelsortclear="Click to clear sorting"
        labelsortdesc="Click to sort Descending"
        perpage="0"
        responsive="sm"
        selectedvariant="primary"
        selectmode="multi"
        small="true"
        sortby=""
        sortdirection="asc"
        striped="true"
        value=""
      />
       
      <template>
        <b-button-stub
          event="click"
          routertag="a"
          tag="button"
          target="_self"
          type="button"
          variant="secondary"
        >
          Back
        </b-button-stub>
      </template>
    </b-card-stub>
  </b-col-stub>
</b-row-stub>
`;
